netrd/distance/resistance_perturbation.py

```diff
@@ -0,0 +1,98 @@
+"""
+resistance_perturbation.py
+--------------
+Graph distance based on resistance perturbation (https://arxiv.org/abs/1605.01091v2)
+
+author: Ryan J. Gallagher & Jessica T. Davis
+email:
+Submitted as part of the 2019 NetSI Collabathon.
+
+"""
+import numpy as np
+import networkx as nx
+import scipy.sparse as ss
+from .base import BaseDistance
+
+class ResistancePerturbation(BaseDistance):
+    def dist(self, G1, G2, p=2):
+        """The resistance perturbation graph distance is the p-norm of the
+        difference between two graph resistance matrices.
+
+        The resistance perturbation distance changes if either graph is relabeled
+        (it is not invariant under graph isomorphism), so node labels should be
+        consistent between the two graphs being compared. The distance is not
+        normalized.
+
+        The resistance matrix of a graph $G$ is calculated as
+        $R = \text{diag}(L_i) 1^T + 1 \text{diag}(L_i)^T - 2L_i$,
+        where L_i is the Moore-Penrose pseudoinverse of the Laplacian of $G$.
+
+        The resistance perturbation graph distance of $G_1$ and $G_2$ is
+        calculated as the $p$-norm of the differenc in their resitance matrices,
+        $d_{r(p)} = ||R^{(1)} - R_^{(2)}|| = [\sum_{i,j \in V} |R^{(1)}_{i,j} - R^{(2)}_{i,j}|^p]^{1/p}$,
+        where R^{(1)} and R^{(2)} are the resistance matrices of $G_1$ and $G_2$
+        respectively. When $p = \infty$,
+        $d_{r(\infty)} = \max_{i,j \in V} |R^{(1)}_{i,j} - R^{(2)}_{i,j}|$.
+
+        For details, see https://arxiv.org/abs/1605.01091v2
+
+        Params
+        ------
+        G1, G2 (nx.Graph): two networkx graphs to be compared.
+        p (float or str, optional): $p$-norm to take of the difference between
+            the resistance matrices. Specify 'infinity' to take $\infty$-norm.
+
+        Returns
+        -------
+        dist (float): the distance between G1 and G2.
+
+        """
+        # Get resistance matrices
+        R1 = get_resistance_matrix(G1)
+        R2 = get_resistance_matrix(G2)
+        self.results['resist1'] = R1
+        self.results['resist2'] = R2
+
+        # Get resistance perturbation distance
+        if p != 'infinity':
```

Comment:
Don't feel too strongly about this, but I don't know how conventional 'infinity' as a string is versus `float('inf')` or `np.inf`.

Reply:
From a user standpoint, I think `float('inf')` and `np.inf` are a little awkward to pass as arguments, especially since they're not needed for the computation itself.

Reply:
Numpy uses `np.inf` for exactly the same parameter `p` (same in spirit, being the dimension of a norm...). See [here](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.linalg.norm.html). I vote we use `np.inf`. I'd also be OK if we used `float('inf')`, but I don't like `'infinity'`.

Reply:
I'm not going to fight strongly against this if you all want to conform with numpy standards, but I just want to make clear that we don't call numpy's norm function. I really don't like `float('inf')`

Reply:
They're [functionally identical](https://stackoverflow.com/questions/42315541/difference-between-np-inf-and-floatinf) and `np.inf` is less obscure than `float('inf')` so I think the numpy approach would be preferred.

Reply:
I know you don't call the numpy norm function, I just meant that it's an example on how to handle a very similar parameter. Can we compromise and use both `np.inf` and `infinity`? If this is too ugly, then I'd very much prefer the use of `np.inf` only. Like Stefan said, we can change this later if needed, so I'm merging this now.